client/src/main/com/sinch/sdk/domains/conversation/adapters/ConversationService.java

```diff
@@ -15,45 +19,55 @@ public class ConversationService implements com.sinch.sdk.domains.conversation.C
   private static final Logger LOGGER = Logger.getLogger(ConversationService.class.getName());
   private static final String SECURITY_SCHEME_KEYWORD_ = "oAuth2";
 
-  private final Configuration configuration;
+  private final String uriUUID;
+  private final ConversationContext context;
   private final HttpClient httpClient;
   private AppService app;
   private ContactService contact;
 
   private final Map<String, AuthManager> authManagers;
 
-  public ConversationService(Configuration configuration, HttpClient httpClient) {
+  public ConversationService(
+      UnifiedCredentials credentials,
+      ConversationContext context,
+      ServerConfiguration oAuthServer,
+      HttpClient httpClient) {
+
+    Objects.requireNonNull(credentials, "Credentials must be defined");
+    Objects.requireNonNull(context, "Context must be defined");
+    StringUtil.requireNonEmpty(credentials.getKeyId(), "'keyId' must be defined");
+    StringUtil.requireNonEmpty(credentials.getKeySecret(), "'keySecret' must be defined");
+    StringUtil.requireNonEmpty(credentials.getProjectId(), "'projectId' must be defined");
+    StringUtil.requireNonEmpty(context.getUrl(), "'url' must be defined");
+    StringUtil.requireNonEmpty(
+        context.getTemplateManagementUrl(), "'templateManagementUrl' must be defined");
```

Comment:
What happens in the case a user is not willing to use the message templates?

Reply:
Currently by using the SinchClient  there is always a value set: https://github.com/sinch/sinch-sdk-java/blob/feat/conversation-rebase-main/client/src/main/com/sinch/sdk/SinchClient.java#L235
But you're right: This will have to refactored when we will have to support context at request level